include/mathfu/matrix.h

```diff
@@ -1360,6 +1388,36 @@ static inline Matrix<T, 4, 4> LookAtHelper(const Vector<T, 3>& at,
 }
 /// @endcond
 
+/// @cond MATHFU_INTERNAL
+///
+template <class T>
+static inline bool UnProjectHelper(const Vector<T, 3>  windowCoord, 
+                                   const Matrix<T, 4, 4> modelView,
+                                   const Matrix<T, 4, 4> projection,
+                                   const float windowWidth,
+                                   const float windowHeight,
+                                   Vector<T, 3>& result) {
+  if (windowCoord.z()<0 || windowCoord.z()>1) {
+    // windowCoord.z should be with in [0, 1]
+    // 0: near plane
+    // 1: far plane
+    return false;
+  }
+  Matrix<T, 4, 4> matrix = (projection * modelView).Inverse();
+  Vector<T, 4> standardized = Vector<T, 4>(2 * (windowCoord.x() - windowWidth) / windowWidth + 1,
+                                           2 * (windowCoord.y() - windowHeight) / windowHeight + 1,
+                                           2 * windowCoord.z() - 1,
+                                           1);
+
+  Vector<T, 4> multiply = matrix * standardized;
+  if(multiply.w() == 0.0) {
```

Comment:
FYI You could setup clang-format with
 
BasedOnStyle=Google
http://clang.llvm.org/docs/ClangFormatStyleOptions.html

to fix the formatting issues